ogr/__init__.py

```diff
@@ -25,22 +25,77 @@
 - simplifying the python work with git
 - intruduce one api for multiple git services (github/gitlab/pagure)
 """
+import logging
+import os
 
 from pkg_resources import get_distribution, DistributionNotFound
 
+from ogr.factory import get_project, get_service_class
+
 try:
     __version__ = get_distribution(__name__).version
 except DistributionNotFound:
     # package is not installed
     pass
 
-from ogr.services.github import GithubService
-from ogr.services.pagure import PagureService
-from ogr.factory import get_project, get_service_class
+logger = logging.getLogger(__name__)
+
+mock_env = os.getenv("MOCK_REQUESTS")
+if mock_env:
+    import github
```

Comment:
> what is reason of imports just inside `if`.

Yes, I would like to completely split the two cases -- mocking/non-mocking



> in ucase you use absolute imports, it should be okay and clean: eg `import ogr.services.mock.tweaks`
> 
> and move these new definitions `def new__requestEncode` there and here just replace it? to avoid bigger code in **init**.py?

I am not sure if I got you right, but definitely, I would like to clean the `__init__` a bit.

Reply:
I've moved as much as possible away. (Some parts cannot be moved because of the import dependencies...)